vllm/core/scheduler.py

```diff
@@ -645,10 +645,14 @@ def _schedule_prefills(
                 assert num_new_tokens == num_prompt_tokens
 
             if num_new_tokens > self.prompt_limit:
+                logger_data = {
```

Comment:
Not sure it's worth it, but we could reduce the number of object allocations if we initialize `logger_data` outside of the `while` loop and recycle the same instance each time `logger_data` is needed. A little clean up of the object would be required after each log call (`logger_data.clear()`), but that could be preferable to more object allocations depending on the nature of the `while` loop.